src/corelibs/U2View/src/ov_sequence/AnnotatedDNAView.cpp

```diff
@@ -736,7 +737,8 @@ void AnnotatedDNAView::removeSequenceWidget(ADVSequenceWidget* sequenceWidget) {
     updateMultiViewActions();
     emit si_sequenceWidgetRemoved(sequenceWidget);
     scrolledWidgetLayout->removeWidget(sequenceWidget);
-    delete sequenceWidget;  // v->deleteLater(); //problem: updates for 'v' after seqCtx is destroyed
+    sequenceWidget->disconnect();
```

Comment:
This may be not enough, because sequenceWidget may contain other widgets that are still connected.
A simpler solution could be the following (delete widget out of the event processing loop)
```
void ADVSingleSequenceWidget::sl_closeView() {
    QTimer::singleShot(0, this, [this] { closeView(); });
}
```

Reply:
I will do the suggested solution. But I don't understand: does your solution change something in terms of "This may be not enough, because sequenceWidget may contain other widgets that are still connected"? In my head, singleShot+closeView doesn't cancel other connections either.

Reply:
> "This may be not enough, because sequenceWidget may contain other widgets that are still connected" 

This is a problem related to your solution: you disconnect some data structures to remove them later, but not all of them (nested structures may still be connected and react to signals when the parent structure is already disconnected).

In the solution I suggested this problem does not exist: you do not disconnect at all: you atomically delete the whole structure.

Reply:
Disconnect is not needed in my solution either. This is an attempt to understand the comment `//problem: updates for 'v' after seqCtx is destroyed`.

Reply:
✓